docs/sql-programming-guide.md

```diff
@@ -733,8 +733,9 @@ SELECT * FROM parquetTable
 
 Table partitioning is a common optimization approach used in systems like Hive. In a partitioned
 table, data are usually stored in different directories, with partitioning column values encoded in
-the path of each partition directory. The Parquet data source is now able to discover and infer
-partitioning information automatically. For example, we can store all our previously used
+the path of each partition directory. All built-in data sources (including TEXT/CSV/JSON/ORC/Parquet)
```

Comment:
`data sources` -> `file sources`
`TEXT` -> `Text`

Reply:
Thanks. It's updated.